mqtt/policy/src/core/mod.rs

```diff
@@ -304,18 +304,8 @@ pub enum Decision {
     Denied,
 }
 
-impl From<Effect> for Decision {
-    fn from(effect: Effect) -> Self {
-        match effect {
-            Effect::Allow => Decision::Allowed,
-            Effect::Deny => Decision::Denied,
-            Effect::Undefined => Decision::Denied,
-        }
-    }
-}
-
 #[derive(Debug, Copy, Clone, PartialOrd, PartialEq)]
-enum Effect {
+enum EffectImpl {
```

Comment:
Not sure I understood this change. You have Effect, EffectImpl, and Decision. What are the differences?

Reply:
`Effect` is a part of policy definition. It can be "undefined" in certain cases. We used to have two `Effect` enums, in mod.rs and `builder.rs`. They are a bit different, but probably can be combined together. I have some ideas I'll explore in the following PR.

`Decision` is a result of policy evaluation. It looks similar, but conceptually a different enum.

Reply:
Can you please put some docs in the comments for each of them covering differences, especially Effect/EffectImpl?